com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs

```diff
@@ -608,86 +597,59 @@ void Swap(ref ScriptableRenderer r)
 
-                    if ((m_Destination.nameID == BuiltinRenderTextureType.CameraTarget && !m_UseSwapBuffer) || (m_ResolveToScreen && m_UseSwapBuffer))
-                        cmd.SetViewport(cameraData.pixelRect);
-
-                    cmd.DrawMesh(RenderingUtils.fullscreenMesh, Matrix4x4.identity, m_Materials.uber);
+                bool setViewProjectionMatrices = true;
+#if ENABLE_VR && ENABLE_XR_MODULE
+                if (cameraData.xr.enabled)
+                    setViewProjectionMatrices = false;
+#endif
 
-                    // TODO: Implement swapbuffer in 2DRenderer so we can remove this
-                    // For now, when render post-processing in the middle of the camera stack (not resolving to screen)
-                    // we do an extra blit to ping pong results back to color texture. In future we should allow a Swap of the current active color texture
-                    // in the pipeline to avoid this extra blit.
-                    if (!m_ResolveToScreen && !m_UseSwapBuffer)
-                    {
-                        cmd.SetGlobalTexture(ShaderPropertyId.sourceTex, cameraTarget);
-                        cmd.SetRenderTarget(m_Source, RenderBufferLoadAction.DontCare, RenderBufferStoreAction.Store, RenderBufferLoadAction.DontCare, RenderBufferStoreAction.DontCare);
-                        cmd.DrawMesh(RenderingUtils.fullscreenMesh, Matrix4x4.identity, m_BlitMaterial);
-                    }
+                if (setViewProjectionMatrices)
+                    cmd.SetViewProjectionMatrices(Matrix4x4.identity, Matrix4x4.identity);
 
-                    cmd.SetViewProjectionMatrices(cameraData.camera.worldToCameraMatrix, cameraData.camera.projectionMatrix);
-                }
+                // TODO: Implement swapbuffer in 2DRenderer so we can remove this
+                // For now, when render post-processing in the middle of the camera stack (not resolving to screen)
+                // we do an extra blit to ping pong results back to color texture. In future we should allow a Swap of the current active color texture
+                // in the pipeline to avoid this extra blit.
+                if (!m_ResolveToScreen && !m_UseSwapBuffer)
+                    Blitter.BlitCameraTexture(cmd, cameraTargetHandle, m_Source, RenderBufferLoadAction.DontCare, RenderBufferStoreAction.Store, m_BlitMaterial, 0);
 
                 if (m_UseSwapBuffer && !m_ResolveToScreen)
                 {
                     renderer.SwapColorBuffer(cmd);
                 }
 
                 // Cleanup
-                if (bloomActive)
-                    cmd.ReleaseTemporaryRT(ShaderConstants._BloomMipUp[0]);
+                if (setViewProjectionMatrices)
+                    cmd.SetViewProjectionMatrices(cameraData.camera.worldToCameraMatrix, cameraData.camera.projectionMatrix);
```

Comment:
If the set identity matrices is removed. We could probably get rid of the matrix "restore" here too.